### PR TITLE
`ImageEditor`: refactor away from `UNSAFE_*`

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* `ImageEditor`: refactor away from `UNSAFE_*`

#### Testing instructions

* Go to `/devdocs/blocks/image-editor`
* Upload an image and verify it's shown in the image editor and controls work as expected
* Upload a new image and verify the image editor is updated correctly

Related to #58453

### CODE REVIEW

File: client/blocks/image-editor/index.jsx [hunk 1]
```diff
@@ -71,22 +71,18 @@ class ImageEditor extends Component {
 
 	editCanvasRef = createRef();
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( newProps ) {
+	componentDidUpdate( prevProps ) {
 		const { media: currentMedia } = this.props;
```
Comment (tyxla) [inline client/blocks/image-editor/index.jsx]: Did you mean to alter `this.props` to `prevProps` here?

File: client/blocks/image-editor/index.jsx [hunk 2]
```diff
@@ -71,22 +71,18 @@ class ImageEditor extends Component {
...
 
-		if ( newProps.media && ! isEqual( newProps.media, currentMedia ) ) {
+		if ( this.props.media && ! isEqual( prevProps.media, currentMedia ) ) {
```
Comment (tyxla) [inline client/blocks/image-editor/index.jsx]: We could use `currentMedia` here.

**Review by tyxla — COMMENTED** [review]
This appears to test well for me, but I have a few notes.

File: client/blocks/image-editor/index.jsx (hunk 1 above)
Comment (flootr) [reply client/blocks/image-editor/index.jsx]: We used to destructure from props before the update but in the next line you can see we're switching usages so this should be fine.

File: client/blocks/image-editor/index.jsx (hunk 2 above)
Comment (flootr) [reply client/blocks/image-editor/index.jsx]: fixed via https://github.com/Automattic/wp-calypso/pull/62944/commits/27ceec7d65c1cd75020ab08c84353474cdda50a8

**Review by tyxla — APPROVED** [review]
LGTM 👍

**Pull request merged**